### DISCUSSION TITLE
TerriaMap WPS

### DISCUSSION THREAD

**Discussion by gtgrp-user** [TerriaJS/terriajs#5942] [Q&A] [answered]
Hi,

I have deployed the TerriaMap version 3.3.3 which works well with the Datacube OWS. I have now deployed the Datacube WPS to test a simple web processing service that returns the average NDVI time series. The web processing service uploads the output in an S3 bucket. When I run the analysis, I can see the Job is running; however, after the Job is finished, I receive the following error message in the console. I checked on S3 the data was uploaded successfully by the process. 

`CatalogFunctionJobMixin.ts?4222:18 Uncaught (in promise) TerriaError {_message: 'Error upserting model JSON: `d4e275b7-3557-46fc-bbeb-303bc0314f51`', _title: {…}, _raisedToUser: false, sender: undefined, severity: ƒ, …}`

Any help would be greatly appreciated! Many thanks.
Pritimoy

**Comment by nf-s** [comment, marked as answer]
Hi @gtgrp-user 

I have created a PR which fixes this issue https://github.com/TerriaJS/terriajs/pull/6030
I will let you know when it has been merged in.

Cheers,
Nick

**Reply by gtgrp-user** [reply]
Thanks very much @nf-s

**Reply by gtgrp-user** [reply]
Hi Nick (@nf-s ),
 
I think issue #6030 is now merged!

Kind  regards
Pritimoy

**Reply by nf-s** [reply]
Yes, sorry I forgot to post in this discussion!

If you are still having issues with v7-v8 conversion, you can set `forceConvertResultsToV8: true` for each `wps` catalog item

**Reply by gtgrp-user** [reply]
Hi Nick,

Thanks for the update. Luckily, I started with the v8.

I have tested the functionality and encountered the following error.

![image](https://user-images.githubusercontent.com/89073412/153579879-0b2a7d0e-8e1d-4dfb-bf0b-fcbe8c462060.png)

Also checked the Water Observation from Space Daily Pixel Drill on the [dev terria cube](https://terria-cube.terria.io/) and the same issue.

![image](https://user-images.githubusercontent.com/89073412/153580137-6e17cbbc-8cf7-477c-8fb5-b15547c69f6b.png)

Kind regards.

Pritimoy

**Reply by nf-s** [reply]
Hi @gtgrp-user 

I made a PR to add the new `forceConvertResultToV8` to the TerriaCube catalog JSON https://github.com/GeoscienceAustralia/dea-config/pull/1016

It has been approved but I don't have write access, so I can't merge the changes

**Reply by gtgrp-user** [reply]
Not a problem, thanks for the update
